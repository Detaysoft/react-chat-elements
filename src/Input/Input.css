.rce-container-input {
    display: flex;
    overflow: hidden;
    min-width: 100%;
    box-sizing: border-box;
    flex-direction: row;
    background: #fff;
    align-items: center;
}

.rce-input {
    flex: 1;
    height: 40px;
    padding: 0 15px;
    border: none;
    border-radius: 5px;
    color: #333;
    font-size: 12px;
    box-sizing: border-box;
    outline: none;
}

.rce-input-textarea {
    height: 25px;
    padding: 5px 15px;
}

.rce-input-buttons {
    display: flex;
    flex-direction: row;
    margin: 5px;
}

.rce-input-buttons > * {
    display: flex;
    flex-direction: row;
}

.rce-input-buttons .rce-button {
    margin-left: 5px;
}

.rce-input-nopadding {
    padding-right: 0px !important;
}

.rce-input::placeholder {
    color: #afafaf;
    font-size: 12px;
}