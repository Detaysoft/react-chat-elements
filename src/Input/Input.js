import React, { Component } from 'react';
import './Input.css';

const classNames = require('classnames');

export class Input extends Component {
	constructor(props) {
		super(props);

		this.state = {
			value: this.props.defaultValue,
		};
	}

	onChange(e) {
		this.setState({
			value: e.target.value
		});
		if (this.props.onChange instanceof Function)
			this.props.onChange(e);

		if (this.props.multiline === true) {
			if (this.props.autoHeight === true) {
				e.target.style.height = this.props.minHeight + 'px';

				if (e.target.scrollHeight <= this.props.maxHeight)
					e.target.style.height = e.target.scrollHeight + 'px';
				else
					e.target.style.height = this.props.maxHeight + 'px';
			}
		}
	}

	render() {
		return (
			<div className="rce-container-input">
				{
					this.props.multiline === false ?
					<input
						type={this.props.type}
						className={classNames("rce-input", {'rce-input-nopadding': this.props.buttons})}
						placeholder={this.props.placeholder}
						value={this.state.value}
						style={this.props.inputStyle}
						onChange={this.onChange.bind(this)}/>
						:
					<textarea
						type={this.props.type}
						className={classNames("rce-input", 'rce-input-textarea', {'rce-input-nopadding': this.props.buttons})}
						placeholder={this.props.placeholder}
						value={this.state.value}
						style={this.props.inputStyle}
						onChange={this.onChange.bind(this)}></textarea>
				}

				{
					this.props.buttons &&
					<div
						className='rce-input-buttons'>
						{this.props.buttons}
					</div>
				}
			</div>
		);
	}
}

Input.defaultProps = {
	type: 'text',
	placeholder: '',
	defaultValue: '',
	onChange: null,
	buttons: null,
	multiline: false,
	minHeight: 25,
	maxHeight: 200,
	autoHeight: true,
	inputStyle: null,
};

export default Input;