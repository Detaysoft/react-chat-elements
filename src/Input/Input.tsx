import React, { useEffect } from 'react'
import './Input.css'
import classNames from 'classnames'
import { IInputProps } from '../type'

const Input: React.FC<IInputProps> = ({
  type = 'text',
  multiline = false,
  minHeight = 25,
  maxHeight = 200,
  autoHeight = true,
  autofocus = false,
  ...props
}) => {
  useEffect(() => {
    if (autofocus === true) props.referance?.current?.focus()

    if (props.clear instanceof Function) {
      props.clear(clear)
    }
  }, [])

  const onChangeEvent = (e: any) => {
    if (props.maxlength && (e.target.value || '').length > props.maxlength) {
      if (props.onMaxLengthExceed instanceof Function) props.onMaxLengthExceed()

      if (props.referance?.current?.value == (e.target.value || '').substring(0, props.maxlength)) return
    }

    if (props.onChange instanceof Function) props.onChange(e)

    if (multiline === true) {
      if (autoHeight === true) {
        if (e.target.style.height !== minHeight + 'px') {
          e.target.style.height = minHeight + 'px'
        }

        let height
        if (e.target.scrollHeight <= maxHeight) height = e.target.scrollHeight + 'px'
        else height = maxHeight + 'px'

        if (e.target.style.height !== height) {
          e.target.style.height = height
        }
      }
    }
  }

  const clear = () => {
    var _event = {
      FAKE_EVENT: true,
      target: props.referance?.current,
    }

    if (props.referance?.current?.value) {
      props.referance.current.value = ''
    }

    onChangeEvent(_event)
  }

  return (
    <div className={classNames('rce-container-input', props.className)}>
      {props.leftButtons && <div className='rce-input-buttons'>{props.leftButtons}</div>}
      {multiline === false ? (
        <input
          ref={props.referance}
          type={type}
          className={classNames('rce-input')}
          placeholder={props.placeholder}
          defaultValue={props.defaultValue}
          style={props.inputStyle}
          onChange={onChangeEvent}
          onCopy={props.onCopy}
          onCut={props.onCut}
          onPaste={props.onPaste}
          onBlur={props.onBlur}
          onFocus={props.onFocus}
          onSelect={props.onSelect}
          onSubmit={props.onSubmit}
          onReset={props.onReset}
          onKeyDown={props.onKeyDown}
          onKeyPress={props.onKeyPress}
          onKeyUp={props.onKeyUp}
          value={props.value}
        />
      ) : (
        <textarea
          ref={props.referance}
          className={classNames('rce-input', 'rce-input-textarea')}
          placeholder={props.placeholder}
          defaultValue={props.defaultValue}
          style={props.inputStyle}
          onChange={onChangeEvent}
          onCopy={props.onCopy}
          onCut={props.onCut}
          onPaste={props.onPaste}
          onBlur={props.onBlur}
          onFocus={props.onFocus}
          onSelect={props.onSelect}
          onSubmit={props.onSubmit}
          onReset={props.onReset}
          onKeyDown={props.onKeyDown}
          onKeyPress={props.onKeyPress}
          onKeyUp={props.onKeyUp}
        >
          {props.defaultValue ? props?.value ?? null : null}
        </textarea>
      )}
      {props.rightButtons && <div className='rce-input-buttons'>{props.rightButtons}</div>}
    </div>
  )
}

export default Input
