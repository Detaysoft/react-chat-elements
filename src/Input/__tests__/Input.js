import React, { Component } from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Input from '../Input'

describe('Input component', () => {
  it('should render without issues', () => {
    const component = shallow(<Input maxHeight={200} />)

    expect(component.length).toBe(1)
    expect(toJson(component)).toMatchSnapshot()
  })
})
