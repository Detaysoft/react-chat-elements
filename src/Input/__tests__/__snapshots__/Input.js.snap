// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render without issues 1`] = `
<div
  className="rce-container-input"
>
  <input
    className="rce-input"
    onChange={[Function]}
    placeholder=""
    style={null}
    type="text"
    value=""
  />
</div>
`;
